src/components/Radio/index.tsx

```diff
@@ -1,145 +1,125 @@
-import { Theme, css } from '@emotion/react'
 import styled from '@emotion/styled'
-import { transparentize } from 'polished'
 import PropTypes from 'prop-types'
-import { InputHTMLAttributes, ReactNode, useMemo } from 'react'
+import { InputHTMLAttributes, ReactNode } from 'react'
 import { Radio as ReakitRadio } from 'reakit'
-import Box, { XStyledProps } from '../Box'
-import Expandable from '../Expandable'
 import Icon from '../Icon'
 import Typography from '../Typography'
 
-const StyledIcon = styled(Icon)``
-
-const IconContainer = styled.div`
-  min-width: 32px;
-  width: 32px;
-  height: 32px;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  margin-right: 8px;
+const StyledIcon = styled(Icon)`
+  margin-right: ${({ theme }) => theme.space['1']};
+  margin-top: ${({ theme }) => theme.space['0.25']};
+  border-radius: ${({ theme }) => theme.radii.circle};
+  fill: ${({ theme }) => theme.colors.neutral.text};
 `
 
-const disabledClass = ({ theme }: { theme: Theme }) => css`
-  color: ${theme.colors.neutral.textDisabled};
-  cursor: not-allowed;
+const StyledRadio = styled(ReakitRadio, {
+  shouldForwardProp: prop => !['size'].includes(prop.toString()),
+})`
+  opacity: 0.01;
+  width: ${({ size }) => size}px;
+  height: ${({ size }) => size}px;
+  position: absolute;
+  cursor: pointer;
 `
 
-const activeFocusClass = ({ theme }: { theme: Theme }) => css`
-  :hover,
-  :focus {
-    ${IconContainer} {
-      background-color: ${transparentize(
-        0.75,
-        theme.colors.primary.backgroundHover,
-      )};
-      border-radius: 50%;
-
-      > ${StyledIcon} {
-        fill: ${theme.colors.primary.textHover};
-      }
-    }
+const StyledRadioContainer = styled(Typography)`
+  position: relative;
+  display: flex;
+  align-items: flex-start;
+
+  &[aria-disabled='false'] {
+    cursor: pointer;
   }
-`
 
-type StyledContainerProps = {
-  disabled: boolean
-  htmlFor: string
-}
+  ${StyledRadio}[aria-checked="true"] + ${StyledIcon} {
+    fill: ${({ theme }) => theme.colors.primary.text};
+  }
 
-const StyledRadioContainer = styled(Typography)<StyledContainerProps>`
-  position: relative;
-  display: flex;
-  align-items: center;
-  height: 32px;
-  cursor: pointer;
+  ${StyledRadio}[aria-invalid="true"] + ${StyledIcon} {
+    fill: ${({ theme }) => theme.colors.danger.text};
+  }
 
-  ${({ disabled }) => (disabled ? disabledClass : activeFocusClass)}
-`
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+    color: ${({ theme }) => theme.colors.neutral.textDisabled};
 
-const StyledRadio = styled(ReakitRadio)`
-  position: absolute;
-  opacity: 0.01;
-`
+    ${StyledIcon} {
+      fill: ${({ theme }) => theme.colors.neutral.textDisabled} !important;
```

Comment:
Please never use `!important`, I get what you want to do here, you should instead put all that disabled css at the bottom of your css so it will be executes last and disabled colors will be set when needed

Reply:
I tried with that but not worked 🫠 i will try to find something

Reply:
done